### PR TITLE
`acc routine` pragmas fix

### PR DESCRIPTION
**Pull request by awnawab**
`acc routine` pragmas should be appended to `Subroutine.spec` rather than prepending to `Subroutine.body`. Although a relatively benign bug, it can cause issues with the pool allocator transformation so I'm filing this bug fix along with updated tests.

### CODE REVIEW

**Review by reuterbal — APPROVED** [review]
Thanks so much! I had recently encountered this as well but couldn't be bothered to file it with a matching test. Much appreciated

**Review by mlange05 — COMMENTED** [review]
All in all this looks great and is much appreciated. I've flagged one corner case that triggered in EC-physics regression, and I'm testing the proposed fix there now. Please change if you agree or ping here if not. Otherwise this is looking read y to me.

File: transformations/transformations/single_column_coalesced.py [hunk 1]
```diff
@@ -481,7 +482,7 @@ def process_kernel(self, routine, demote_locals=True):
             Subroutine to apply this transformation to.
         """
 
-        pragmas = FindNodes(ir.Pragma).visit(routine.body)
+        pragmas = FindNodes(ir.Pragma).visit(routine.spec)
```
[the remark this answers is not in the document]
Comment (awnawab) [reply transformations/transformations/single_column_coalesced.py]: Yes sure, thanks for catching this, done!

**Review by mlange05 — APPROVED** [review]
Ok, thanks for addressing this. Confirmed clean with EC-physics regression now, so GTG from me! :shipit:
